File: Company_wise_Mock/Facebook/Range Sum of BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
        if not root:
            return 0
        
        cnt = 0
        if low <= root.val <= high:
            cnt += root.val
        
        cnt += self.rangeSumBST(root.left, low, high) + self.rangeSumBST(root.right, low, high)
        return cnt

File: Company_wise_Mock/Facebook/Range Sum of BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int ans = 0;
    void solve(TreeNode* root, int low, int high)
    {
        if(!root) return;
        
        if(root->val >= low && root->val <= high)
            ans += root->val;
        
        solve(root->left, low, high);
        solve(root->right, low, high);
    }
    
    int rangeSumBST(TreeNode* root, int low, int high) {
        solve(root, low, high);
        return ans;
    }
};
